src/presets/firebase.ts

```diff
@@ -65,17 +65,19 @@ async function writeRoutes(nitro: Nitro) {
     return obj;
   }, {} as Record<string, string>);
 
-  let nodeVersion = "14";
+  let nodeVersion = "18";
+  const supportedNodeVersions = ["18", "16", "14", "12", "10"];
+  //    ^ See https://cloud.google.com/functions/docs/concepts/nodejs-runtime
   try {
     const currentNodeVersion = JSON.parse(
       await readFile(join(nitro.options.rootDir, "package.json"), "utf8")
     ).engines.node;
```

Comment:
Related: engines.node could be like `18 || 14` too. We might parse it as array and choose first matching version.

Reply:
Could be in later PR..